emcc.py

```diff
@@ -1302,6 +1302,10 @@ def get_bitcode_args(input_files):
           args += shared.EMSDK_CXX_OPTS
         if not shared.Building.can_inline():
           args.append('-fno-inline-functions')
+        # For fastcomp backend, no LLVM IR functions should ever be annotated 'optnone', because that would skip running the SimplifyCFG pass on them, which is required to always run to
+        # clean up LandingPadInst instructions that are not needed.
+        if not shared.Settings.WASM_BACKEND:
+          args += ['-Xclang', '-disable-O0-optnone']
```

Comment:
Are you saying that llvm changed semantics so that the SimplifyCFG pass will not run on optnone methods now? Or is optnone a new thing in this merge?

Perhaps there are other consequences to not doing optnone? Like slower compile times for -O0?

Reply:
> Are you saying that llvm changed semantics so that the SimplifyCFG pass will not run on optnone methods now?

That semantic has not changed, all optimization passes have a test for `if (func is optnone) skip this pass on this function;`

> Or is optnone a new thing in this merge?

The new thing is that the `optnone` attribute is now getting slapped to all functions when compiling code with `-O0` optimization level, whereas it used to not be added. When this change was done, the `-disable-O0-optnone` flag was added to allow reverting to the earlier behavior of not producing all functions with `optnone`.

Reply:
This does still worry me, but the only concrete concern I have is compile times. Can you please check if -O0 compile times vary much with and without that flag? (on a file where it doesn't crash the compiler to not run that pass that optnone disables).

Reply:
Check, running timing benchmarks now

Reply:
Running `default` suite with `-disable-O0-optnone` present gives 1367 seconds, and running `default` suite without `-disable-O0-optnone` gives 1303 seconds (-4.68%), though that suite fails on 7 tests, so not sure if that has a bearing.

However, I don't think it's fair to compare compile times to the version where the `optnone` flag is present, since that is not the original status quo - we have never generated code with that attribute applied to all functions. I.e. we used to behave as if `optnone` did not exist. See this commit that introduced the `-disable-O0-optnone` flag: https://github.com/juj/emscripten-fastcomp-clang/commit/7a4550308bb4475544102bf06dc6b64297aaa014 It is that commit which also introduced the sprinkling of `optnone` attribute everywhere in `-O0`.

Reply:
Running a comparison against pre-merge branch next.

Reply:
Ran `default` suite in current `incoming` branch, getting 1433 seconds as runtime. So at least the update will not regress overall compilation performance, but improves it by a little (-4.61% less time).

Reply:
Thanks, that seems ok then.